### DISCUSSION TITLE
Exporting styled-system with design system

### DISCUSSION THREAD

**Discussion by benoitdeziel** [chakra-ui/panda#1023] [Q&A] [answered]
Hello! First of all thank you so much for this amazing library. The DX is incredible and I can't wait to migrate our design system to Panda.

I'm currently doing a POC to promote the idea to switch to Panda. I followed the component library instructions in the docs but I still have some questions. Would it be possible to do `export * from '@company/styled-system` from our design system (considering I'm using `emitPackage: true`) so that the developer using the ds to only do `import { css , cva, cx } from '@company/ds'`? 

Everything is working fine in an app with the `outdir` set to `@company/styled-system` and `emitPackage: true` but I would like to have only an import from the main ds and not from another package to use some functions/recipes/patterns.

This might be only an issue with my POC since I'm using NX with PNPM, with apps and ds in the same repository. In a real world scenario, the ds is separated from the apps monorepo in it's own repo an published to NPM.

**Comment by astahmer** [comment, marked as answer]
if you only generate 1 outdir for all your packages (`import { css , cva, cx } from '@company/ds'`), it should be generated using a `config.include` that includes all your sources, which is probably not what you want

---

if you have multiple packages each with different config `include`, you probably want to generate an `outdir` for each of them so that the lightweight JS runtime is not duplicated (`css` and `cva` function, patterns, recipes, jsx components..) and the css generated matches only the `include` for that package

by using `emitPackage: true` and re-using the same `outdir` name for each package, you are simulating (by tricking node resolution tbh) having only 1 import
